From f66e47baf938c477653a555e8ccf60b5f4a7dbd9 Mon Sep 17 00:00:00 2001
From: Mikhail Ulyanov <mikhail.ulyanov@cogentembedded.com>
Date: Tue, 19 Aug 2014 16:50:51 +0400
Subject: [PATCH 23/25] ARM: shmobile: r8a7791: Add JPU clock dt and CPG
 define.

Signed-off-by: Mikhail Ulyanov <mikhail.ulyanov@cogentembedded.com>
Acked-by: Laurent Pinchart <laurent.pinchart@ideasonboard.com>
Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
(cherry picked from commit ed48b5d6fd339d145df5a6a1e48cf56ef265cf4f)
Signed-off-by: Simon Horman <horms+renesas@verge.net.au>
Signed-off-by: Andrey Gusakov <andrey.gusakov@cogentembedded.com>
---
 arch/arm/boot/dts/r8a7791.dtsi            |    6 +++---
 include/dt-bindings/clock/r8a7791-clock.h |    1 +
 2 files changed, 4 insertions(+), 3 deletions(-)

diff --git a/arch/arm/boot/dts/r8a7791.dtsi b/arch/arm/boot/dts/r8a7791.dtsi
index b2506c5..b3f0eb9 100644
--- a/arch/arm/boot/dts/r8a7791.dtsi
+++ b/arch/arm/boot/dts/r8a7791.dtsi
@@ -882,20 +882,20 @@
 		mstp1_clks: mstp1_clks@e6150134 {
 			compatible = "renesas,r8a7791-mstp-clocks", "renesas,cpg-mstp-clocks";
 			reg = <0 0xe6150134 0 4>, <0 0xe6150038 0 4>;
-			clocks = <&zs_clk>, <&zs_clk>, <&zs_clk>, <&p_clk>,
+			clocks = <&zs_clk>, <&zs_clk>, <&m2_clk>, <&zs_clk>, <&p_clk>,
 				 <&zg_clk>, <&zs_clk>, <&zs_clk>, <&zs_clk>,
 				 <&p_clk>, <&p_clk>, <&rclk_clk>, <&cp_clk>,
 				 <&zs_clk>, <&zs_clk>, <&zs_clk>;
 			#clock-cells = <1>;
 			renesas,clock-indices = <
-				R8A7791_CLK_VCP0 R8A7791_CLK_VPC0 R8A7791_CLK_SSP1
+				R8A7791_CLK_VCP0 R8A7791_CLK_VPC0 R8A7791_CLK_JPU R8A7791_CLK_SSP1
 				R8A7791_CLK_TMU1 R8A7791_CLK_PVRSRVKM R8A7791_CLK_2DDMAC
 				R8A7791_CLK_FDP1 R8A7791_CLK_FDP0 R8A7791_CLK_TMU3
 				R8A7791_CLK_TMU2 R8A7791_CLK_CMT0 R8A7791_CLK_TMU0
 				R8A7791_CLK_VSP1_DU1 R8A7791_CLK_VSP1_DU0 R8A7791_CLK_VSP1_S
 			>;
 			clock-output-names =
-				"vcp0", "vpc0", "ssp_dev", "tmu1", "pvrsrvkm",
+				"vcp0", "vpc0", "jpu", "ssp_dev", "tmu1", "pvrsrvkm",
 				"tddmac", "fdp1", "fdp0", "tmu3", "tmu2",
 				"cmt0", "tmu0", "vsp1-du1", "vsp1-du0", "vsps";
 		};
diff --git a/include/dt-bindings/clock/r8a7791-clock.h b/include/dt-bindings/clock/r8a7791-clock.h
index 0327180..292148d 100644
--- a/include/dt-bindings/clock/r8a7791-clock.h
+++ b/include/dt-bindings/clock/r8a7791-clock.h
@@ -27,6 +27,7 @@
 /* MSTP1 */
 #define R8A7791_CLK_VCP0		1
 #define R8A7791_CLK_VPC0		3
+#define R8A7791_CLK_JPU			6
 #define R8A7791_CLK_SSP1		9
 #define R8A7791_CLK_TMU1		11
 #define R8A7791_CLK_PVRSRVKM		12
-- 
1.7.10.4

